pkg/scheduler/core/spreadconstraint/select_clusters_by_region.go

```diff
@@ -35,66 +36,131 @@ func selectBestClustersByRegion(spreadConstraintMap map[policyv1alpha1.SpreadFie
 	}
 
 	// thirdly, select the remaining Clusters based cluster.Score
-	sortClusters(candidateClusters)
 	restCnt := needCnt - len(clusters)
-	for i := 0; i < restCnt; i++ {
-		clusters = append(clusters, candidateClusters[i].Cluster)
+	if restCnt > 0 {
+		sortClusters(candidateClusters)
+		for i := 0; i < restCnt; i++ {
+			clusters = append(clusters, candidateClusters[i].Cluster)
+		}
 	}
 
 	return clusters, nil
 }
 
-func selectRegions(RegionInfos map[string]RegionInfo, regionConstraint, clusterConstraint policyv1alpha1.SpreadConstraint) []RegionInfo {
-	var regions []RegionInfo
-	for i := range RegionInfos {
-		regions = append(regions, RegionInfos[i])
+// getSortedRegions returns sorted region slice by score and available replicas.
+func getSortedRegions(regionInfoMap map[string]RegionInfo) []RegionInfo {
+	regions := make([]RegionInfo, 0, len(regionInfoMap))
+	for _, region := range regionInfoMap {
+		regions = append(regions, region)
 	}
 
 	sort.Slice(regions, func(i, j int) bool {
 		if regions[i].Score != regions[j].Score {
 			return regions[i].Score > regions[j].Score
```

Comment:
By the way, currently the region score is the highest score within the cluster. I think this algorithm may not reflect the overall situation of this regional cluster very well. Would it be more reasonable to use median, average or some other algorithm instead?

Reply:
Yeah, I've done this [optimization,](https://github.com/karmada-io/karmada/pull/3259/files#diff-d71362f4d2ac9a200a70291dd5117b7c6e3bc9de9880a23dc536361606ad3ad5R136) it does not conflict with your PR(https://github.com/karmada-io/karmada/pull/3287). Can you take a look?